src/SSHDebugPS/OutputWindow.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.SSHDebugPS
+{
+    class OutputWindow
+    {
+        private static OutputWindow _instance;
+
+        public static OutputWindow Instance
+        {
+            get
+            {
+                if (_instance == null)
+                {
+                    _instance = new OutputWindow();
+                }
+
+                return _instance;
+            }
+        }
+
+        private class PaneInfo
+        {
+            public PaneInfo(Guid paneId)
+            {
+                this.paneId = paneId;
+                this.Shown = false;
+            }
+
+            internal Guid paneId;
+            internal bool Shown { get; set; }
+        }
+
+        private IVsOutputWindow outputWindow;
+        private IVsUIShell shell;
+
+        private Dictionary<string, PaneInfo> panes = new Dictionary<string, PaneInfo>()
+        {
+            // The 'Debug' pane exists by default
+            { "Debug", new PaneInfo(VSConstants.GUID_OutWindowDebugPane) }
+        };
+
+        private OutputWindow()
+        {
+            outputWindow = Package.GetGlobalService(typeof(SVsOutputWindow)) as IVsOutputWindow;
```

Comment:
if this runs outside of VS (like during tests) this will throw. Think about putting it in a try/catch.

Reply:
Make sure to run glass tests. You may actually need to move these calls to a separate class to be able to catch the failure.

---
In reply to: [452387606](https://github.com/microsoft/MIEngine/pull/1017#discussion_r452387606) [](ancestors = 452387606)

Reply:
Actually, wait, we don't have glass tests for the SSH port supplier - just the engine.

---
In reply to: [452411351](https://github.com/microsoft/MIEngine/pull/1017#discussion_r452411351) [](ancestors = 452411351,452387606)